components/formats-gpl/src/loci/formats/in/CV7000Reader.java

```diff
@@ -0,0 +1,900 @@
+              // There was likely an error during acquisition for this
+              // particular plane.  Skip it.
+              continue;
+            }
+            Channel channel = null;
+            for (Channel ch : channels) {
+              if (ch.index == p.channel) {
+                channel = ch;
+                break;
+              }
+            }
+            if (channel == null) {
+              continue;
+            }
+
+            if (c == 0) {
+              store.setPixelsPhysicalSizeX(FormatTools.getPhysicalSizeX(channel.xSize), i);
+              store.setPixelsPhysicalSizeY(FormatTools.getPhysicalSizeY(channel.xSize), i);
+            }
+
+            int objective = -1;
+            if (channel.objectiveID != null) {
+              objective = usedObjectiveIDs.indexOf(channel.objectiveID);
+            }
+
+            if (channel.magnification != null && objective >= 0) {
+              store.setObjectiveNominalMagnification(channel.magnification, 0, objective);
+            }
+            if (objective >= 0) {
+              String objectiveID = MetadataTools.createLSID("Objective", 0, objective);
+              store.setObjectiveSettingsID(objectiveID, i);
+            }
+
+            store.setChannelName("Channel #" + (channel.index + 1) + ", Camera #" + channel.cameraNumber, i, c);
+
+            if (channel.color != null) {
+              store.setChannelColor(channel.color, i, c);
+            }
+
+            if (channel.excitation != null && channel.lightSourceRefs != null) {
+              int index = -1;
+              for (int ref=0; ref<channel.lightSourceRefs.size(); ref++) {
+                int lightSource = channel.lightSourceRefs.get(ref);
+                if ("Laser".equals(lightSources.get(lightSource).type) &&
+                  lightSources.get(lightSource).wavelength < channel.excitation)
+                {
+                  index = lightSource;
+                }
+              }
+              if (index >= 0) {
+                store.setChannelLightSourceSettingsID(
+                  MetadataTools.createLSID("LightSource", 0, index), i, c);
+                store.setChannelExcitationWavelength(
+                  new Length(channel.excitation, UNITS.NM), i, c);
+              }
+            }
+
+            if (channel.exposureTime != null) {
+              Time exposure = new Time(channel.exposureTime, UNITS.MS);
+              for (int z=0; z<getSizeZ(); z++) {
+                for (int t=0; t<getSizeT(); t++) {
+                  int plane = getIndex(z, c, t);
+                  store.setPlaneExposureTime(exposure, i, plane);
+                }
+              }
+            }
+          }
+        }
+
+        for (int p=0; p<getImageCount(); p++) {
+          Plane plane = lookupPlane(i, p);
+          if (plane == null) {
+            continue;
+          }
+          store.setPlanePositionX(FormatTools.createLength(plane.xpos, UNITS.REFERENCEFRAME), i, p);
```

Comment:
Similarly to #3244, we might want to fill the corresponding `WellSample.Position` metadata with the same values when possible. This can happen as a follow-up PR.

Reply:
Noted for follow-up PR: https://trello.com/c/b0nFSUzZ/49-cv7000#comment-5bc498c81ae2b885ae77c111